### PR TITLE
A few suggestions

### PR DESCRIPTION
**Pull request by danmactough**
Happy to kill any of these you don't like, but it's easier to see them all together than individual branches and PRs. Each commit is one suggestion.

I believe a600b94 is the correct answer to #5.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -2,60 +2,77 @@ var dir = require("node-dir");
...
 var fs = require("fs");
 
-module.exports = function(folder, cb){
+module.exports = function(folder, cb) {
 
-	dir.files(folder, function(err, files) {
-		if(err){
-			cb(err);
+	var cwd = process.cwd();
+	process.chdir(folder);
```
Comment (mcwhittemore) [inline index.js:8]: Why do we need to change directory?

File: index.js [hunk 2]
```diff
@@ -2,60 +2,77 @@ var dir = require("node-dir");
...
-					if(bits[1] == "node_modules"){
-						filesByModule[bits[2]] = filesByModule[bits[2]] || [];
-						filesByModule[bits[2]].push(files[i]);
+			for (var i = 0; i < files.length; i++) {
+				var file = files[i];
+				var fileParts = file.split(path.sep);
+				var lastIdx = fileParts.lastIndexOf("node_modules");
+
+				if (file.match(/\.js$/)) {
+					if (lastIdx >= 0) {
```
Comment (mcwhittemore) [inline index.js:31]: Are you doing this so it can be run in a node_modules folder?

File: index.js [hunk 3]
```diff
@@ -2,60 +2,77 @@ var dir = require("node-dir");
...
+						filesByModule[moduleName] = filesByModule[moduleName] || [];
+						filesByModule[moduleName].push(files[i]);
+						if (moduleNames.indexOf(moduleName) === -1) {
+							moduleNames.push(moduleName);
+						}
 					}
-					else{
-						filesByModule["active"].push(files[i]);
+					else {
+						filesByModule["main"].push(files[i]); // eslint-disable-line dot-notation
```
Comment (mcwhittemore) [inline index.js]: `filesByModule.main.push(files[i]);`

If we're going to have the eslint we might as well use it :)

File: index.js (hunk 2 above)
Comment (danmactough) [inline index.js:31]: nested node_modules, e.g., for file `node_modules/foo/node_modules/bar/node_modules/baz/index.js`

before, module name was detected as "foo" - now, detected as "baz"

File: index.js (hunk 1 above)
Comment (danmactough) [inline index.js:8]: it eliminates the need to do `var file = files[i].replace(folder, "");` on every filename, which in addition to being unnecessary (if you chdir), is kind of hacky, which always worries me.

File: index.js (hunk 3 above)
Comment (danmactough) [inline index.js]: :+1:

Comment (danmactough) [inline index.js]: fixed 3674795

File: index.js (hunk 1 above)
Comment (mcwhittemore) [inline index.js:8]: I don't like the idea of scripts changing my directory. I'm not sure which is the right practice. If I have two scripts running with & what is the effect?

File: index.js (hunk 2 above)
Comment (mcwhittemore) [inline index.js:31]: +1

File: index.js (hunk 1 above)
Comment (danmactough) [inline index.js:8]: It's just the process. Pretty common for a shell script, which is basically what this is doing. I change it back when I'm done, but even if I didn't, when the process ends, you'd be back in the original directory.

**Pull request merged**